### PR TITLE
Add waitForPlan and similar functions

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Fix test failures in resilience_repair by assuring the last plan version the tests wrote is the one being repaired.

Only affects the tests, this is not a problem for productive usage.

Fixes https://arangodb.atlassian.net/browse/BTS-59.

- [X] Bug-Fix for *devel-branch*
- [X] Bug-Fix for *3.7* (no need for further backports, problem was introduced by incremental agency sync): https://github.com/arangodb/arangodb/pull/11916
- [X] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as the `resilience_repair` suite.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10393/

File: arangod/Cluster/ClusterInfo.cpp [hunk 1]
```diff
@@ -1175,13 +1179,19 @@ void ClusterInfo::loadPlan() {
   {
     std::lock_guard w(_waitPlanLock);
     triggerWaiting(_waitPlan, idx);
+    // TODO Wouldn't it be better to notify only when the plan version changes?
```
Comment (maierlars) [inline arangod/Cluster/ClusterInfo.cpp:1182]: Follower DBServer look at the Current entry to determine if they have to initalize a synchronizeShard action. So I would say, no.

Comment (maierlars) [reply arangod/Cluster/ClusterInfo.cpp:1182]: So the above triggers Futures that wait for _any_ update no matter what it caused?

Comment (goedderz) [reply arangod/Cluster/ClusterInfo.cpp:1182]: I'll remove the comment, I think its fine.

`triggerWaiting` triggers only futures waiting for a raft index not newer than the one we saw here.

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10438/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10462/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10480/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/10485/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/10485/

**Comment by goedderz** [comment]
image

**Pull request merged**